src/Umbraco.Infrastructure/Security/MemberRoleStore.cs

```diff
@@ -0,0 +1,293 @@
+    {
+        private readonly IMemberGroupService _memberGroupService;
+        private bool _disposed;
+
+        //TODO: Move into custom error describer.
+        //TODO: How revealing can the error messages be?
+        private readonly IdentityError _intParseError = new IdentityError { Code = "IdentityIdParseError", Description = "Cannot parse ID to int" };
+        private readonly IdentityError _memberGroupNotFoundError = new IdentityError { Code = "IdentityMemberGroupNotFound", Description = "Member group not found" };
+        private const string genericIdentityErrorCode = "IdentityErrorUserStore";
+
+        public MemberRoleStore(IMemberGroupService memberGroupService, IdentityErrorDescriber errorDescriber)
+        {
+            _memberGroupService = memberGroupService ?? throw new ArgumentNullException(nameof(memberGroupService));
+            ErrorDescriber = errorDescriber ?? throw new ArgumentNullException(nameof(errorDescriber));
+        }
+
+        /// <summary>
+        /// Gets or sets the <see cref="IdentityErrorDescriber"/> for any error that occurred with the current operation.
+        /// </summary>
+        public IdentityErrorDescriber ErrorDescriber { get; set; }
+
+        /// <inheritdoc />
+        public Task<IdentityResult> CreateAsync(TRole role, CancellationToken cancellationToken = default)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            ThrowIfDisposed();
+
+            if (role == null)
+            {
+                throw new ArgumentNullException(nameof(role));
+            }
+
+            var memberGroup = new MemberGroup
+            {
+                Name = role.Name
+            };
+
+            _memberGroupService.Save(memberGroup);
+
+            role.Id = memberGroup.Id.ToString();
+
+            return Task.FromResult(IdentityResult.Success);
+        }
+
+
+        /// <inheritdoc />
+        public Task<IdentityResult> UpdateAsync(TRole role, CancellationToken cancellationToken = default)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            ThrowIfDisposed();
+
+            if (role == null)
+            {
+                throw new ArgumentNullException(nameof(role));
+            }
+
+            if (!int.TryParse(role.Id, out int roleId))
+            {
+                return Task.FromResult(IdentityResult.Failed(_intParseError));
+            }
+
+            IMemberGroup memberGroup = _memberGroupService.GetById(roleId);
+            if (memberGroup != null)
+            {
+                if (MapToMemberGroup(role, memberGroup))
+                {
+                    _memberGroupService.Save(memberGroup);
+                }
+                //TODO: if nothing changed, do we need to report this?
+                return Task.FromResult(IdentityResult.Success);
+            }
+            else
+            {
+                //TODO: throw exception when not found, or return failure?
```

Comment:
Its a good question and seems that the Entity Framework implementation doesn't specifically deal with this  but seems to indicate that no exceptions are thrown if something fails. https://github.com/dotnet/aspnetcore/blob/main/src/Identity/EntityFrameworkCore/src/RoleStore.cs#L157

So I think we stick to what you have with the Failed approach.